src/pages/index.js

```diff
@@ -38,30 +94,80 @@ cardAddButton.addEventListener("click", () => {
   cardAddPopup.open();
 });
 
+avatarEditBtn.addEventListener("click", () => {
+  avatarEditPopup.open();
+});
+
 function renderCard(cardData) {
   const newCard = getCardElement(cardData);
   cardList.addItem(newCard);
 }
 
-const cardList = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
-);
+const cardList = new Section({ renderer: renderCard }, ".cards__list");
 
-cardList.renderItems();
+api
+  .getAppInfo()
+  .then(([userData, cards]) => {
+    cardList.renderItems(cards);
+    profileUserInfo.setUserInfo(userData);
+    profileUserInfo.setUserAvatar(userData);
+  })
+  .catch((error) => console.error(error));
 
 const cardAddPopup = new PopupWithForm({
   popupSelector: "#card-add-modal",
   handleFormSubmit: (data) => {
-    renderCard(data);
-    cardFormValidator.disableButton();
+    renderLoading("#card-add-modal", true);
```

Comment:
**COULD BE IMPROVED**

It’s better to make a `renderLoading` method inside `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```